BeagleBone_black/src/communications/communications.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Authors: Kofi and Isabella
+ * Organisation: HYPED
+ * Date: 1. April 2018
+ * Description:
+ *
+ *    Copyright 2018 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "communications.hpp"
+
+
+#include <sstream>
+#include <string>
+
+namespace hyped {
+
+using data::Communications;
+
+namespace communications {
+
+Communications::Communications(Logger& log, const char* ip, int portNo)
+    : log_(log)
+{
+  log_.INFO("COMN", "BaseCommunicator initialised.");
+  sockfd_ = socket(AF_INET, SOCK_STREAM, 0);   // socket(int domain, int type, int protocol)
+
+  if (sockfd_ < 0) {
+    log_.ERR("COMN", "CANNOT OPEN SOCKET.");
+  }
+
+  server = gethostbyname(ip);
+
+  if (server == NULL) {
+    log_.ERR("COMN", "INCORRECT BASE-STATION IP, OR BASE-STATION S/W NOT RUNNING.");
+  }
+
+  memset(&serv_addr, '\0', sizeof(serv_addr));
+  serv_addr.sin_family = AF_INET;   // server byte order
+  memcpy(server->h_addr, &serv_addr.sin_addr.s_addr, server->h_length);
+  serv_addr.sin_port = htons(portNo);
+
+  if (connect(sockfd_, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
+    log_.ERR("COMN", "CANNOT ESTABLISH CONNECTION TO BASE-STATION.");
+  }
+
+  log_.INFO("COMN", "TCP/IP connection established.");
+}
+
+Communications::~Communications()
+{
+  close(sockfd_);
+}
+
+int Communications::sendDistance(float distance)
+{
+  stringstream ss(stringstream::in | stringstream::out);
+  ss << distance;
```

Comment:
Are you only using streams to convert numbers to strings? If so, consider using [`to_string()` method](http://www.cplusplus.com/reference/string/to_string/). I don't have experience with streams but I would expect they are only really useful when you need to concatenate a lot of strings.